src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splittedDate = date.split(fromFormat[fromFormat.length - 1]);
+  const dateProperties = {};
+  const updatedDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    dateProperties[fromFormat[i]] = splittedDate[i];
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateProperties['YY'] = dateProperties['YYYY'].slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (dateProperties.YY < 30) {
+      dateProperties['YYYY'] = '20' + dateProperties['YY'];
+    } else {
+      dateProperties['YYYY'] = '19' + dateProperties['YY'];
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
+    updatedDate.push(dateProperties[toFormat[i]]);
```

Comment:
The loop for constructing the new date format is clear and aligns with the given task. The use of `push` to build the `updatedDate` array is appropriate here.